test/test_datasets.py

```diff
@@ -29,6 +41,190 @@ def test_get_intersection_range(self):
         s2 = pd.Series(pd.date_range('2020-10-01', '2020-12-01'))
         self._test_get_intersection_range(s1, s2, 32)
 
+    @patch('pyronear_ds.datasets.utils.requests.get')
+    def test_url_retrieve(self, mock_get):
+
+        mock_get.return_value.status_code = 200
+        mock_get.return_value.content = bytes("WEATHER OR WILDFIRE FILE", 'utf-8')
+        content = utils.url_retrieve('url')
+        self.assertIsInstance(content, bytes)
+
+        mock_get.return_value.status_code = 400
+        mock_get.return_value.content = bytes("WEATHER OR WILDFIRE FILE", 'utf-8')
+        self.assertRaises(requests.exceptions.ConnectionError, utils.url_retrieve, 'url')
+
+    def test_get_fname(self):
+
+        url_firms = "https://firms.modaps.eosdis.nasa.gov/data/active_fire/c6/csv/MODIS_C6_Europe_24h.csv"
+        url_ghcn = "https://www1.ncdc.noaa.gov/pub/data/ghcn/daily/by_year/2020.csv.gz"
+        url_isd = "https://www.ncei.noaa.gov/data/global-hourly/archive/csv/2020.tar.gz"
+
+        self.assertEqual(utils.get_fname(url_firms), ("MODIS_C6_Europe_24h", "csv", None))
+        self.assertEqual(utils.get_fname(url_ghcn), ("2020", "csv", "gz"))
+        self.assertEqual(utils.get_fname(url_isd), ("2020", None, "tar.gz"))
+
+    @staticmethod
+    def _make_tarfile(destination):
+
+        unzipped_content = [
+            ['col1', 'col2', 'col3', 'col4'],
+            ['test', 'test', 'test', 'test'],
+            ['test', 'test', 'test', 'test'],
+            ['test', 'test', 'test', 'test']]
+
+        full_path = os.path.join(destination, 'server/')
+
+        os.makedirs(os.path.dirname(full_path), exist_ok=True)
+
+        with open(os.path.join(full_path, "test_tar.csv"), mode='w') as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerows(unzipped_content)
+
+        out = tarfile.open(os.path.join(full_path, "test.tar.gz"), 'w:gz')
+        out.add(full_path, arcname=os.path.basename(full_path))
+        out.close()
+
+        with open(os.path.join(full_path, "test.tar.gz"), 'rb') as tar_file:
+            memory_file = BytesIO(tar_file.read())
+
+        return memory_file
```

Comment:
Most of this block is duplicated 3 extra times, could you try to come up with a light refactoring please? :pray:

Reply:
Thanks for the reviews, I will improve that !